core/server/models/base/index.js

```diff
@@ -394,17 +394,20 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
     },
 
     // Get the user from the options object
-    contextUser: function contextUser(options) {
-        options = options || {};
-        options.context = options.context || {};
+    contextUser: function contextUser(options = {}) {
+        const context = options.context || {};
+
+        if (context.api_key_id && context.api_key_user && context.api_key_user.get('id')) {
+            return context.api_key_user.get('id');
```

Comment:
What was the reason to not re-use `context.user`? 🤔

Reply:
Because `context.user` has much larger implications, all the existing code assumes that the request is being made by a user if `context.user` exists. So we could easily introduce an unintended side effect. Also `context.user` currently _only_ holds the id, and if we were to overload it with a model as well I think it would start to get confusing.